src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```
@@ -29,6 +29,10 @@ Copyright (c) .NET Foundation. All rights reserved.
   <UsingTask TaskName="GetAssemblyVersion" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
   <UsingTask TaskName="GenerateSatelliteAssemblies" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
 
+  <ItemGroup Condition="'$(BuildWithFrameworkRoslyn)' == 'true' and '$(MicrosoftNetFrameworkCompilersToolsetPackageVersion)' != '' and @(PackageReference->AnyHaveMetadataValue('Identity', 'Microsoft.Net.Compilers.Toolset.Framework')) != 'true'">
```

Check if PackageReference is not empty as otherwise the `AnyHaveMetadataValue` msbuild function would throw an exception. Also no need to do the `!= true`.

```suggestion
  <ItemGroup Condition="'$(BuildWithFrameworkRoslyn)' == 'true' and '$(MicrosoftNetFrameworkCompilersToolsetPackageVersion)' != '' and ('@(PackageReference)' == '' or  !@(PackageReference->AnyHaveMetadataValue('Identity', 'Microsoft.Net.Compilers.Toolset.Framework')))">
```

---

This is the case where there are no package references?

---

Yes. If no package references exist, the `AnyHaveMetadataValue` msbuild function would fail. That's an unfortunate behavior with msbuild item functions that return a boolean.

---

I put in a fix specifically for AnyHaveMetadataValue to make it return the proper value even if there are no PackageReferences. Prior to that, it would just notice an empty list and return an empty list. != true handles both cases.